clang/include/clang/Basic/DiagnosticIDs.h

```diff
@@ -200,7 +273,33 @@ class DiagnosticIDs : public RefCountedBase<DiagnosticIDs> {
   // FIXME: Replace this function with a create-only facilty like
   // createCustomDiagIDFromFormatString() to enforce safe usage. At the time of
   // writing, nearly all callers of this function were invalid.
-  unsigned getCustomDiagID(Level L, StringRef FormatString);
+  unsigned getCustomDiagID(CustomDiagDesc Diag);
+
+  [[deprecated("Use a CustomDiagDesc instead of a Level")]] unsigned
+  getCustomDiagID(Level Level, StringRef Message) {
+    return getCustomDiagID([&]() -> CustomDiagDesc {
+      switch (Level) {
+      case DiagnosticIDs::Level::Ignored:
+        return {diag::Severity::Ignored, std::string(Message), CLASS_WARNING,
```

Comment:
If it's ignored, why is it class warning and should be shown in system headers?

Actually, why are all of these saying to show in system headers? Was that the previous behavior and we're just matching it?

Reply:
Yes, this is just matching the previous behaviour. I agree that it doesn't make a ton of sense in some cases, but this should go at some point anyways, so I didn't want to mess with it.